test_runner/src/main/kotlin/ftl/reports/api/PerformanceMetrics.kt

```diff
@@ -3,25 +3,39 @@ package ftl.reports.api
 import com.google.api.services.testing.model.TestExecution
 import com.google.api.services.toolresults.model.PerfMetricsSummary
 import ftl.android.AndroidCatalog
+import ftl.args.IArgs
 import ftl.gc.GcStorage
 import ftl.gc.GcToolResults
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.async
 import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.runBlocking
+import java.nio.file.Files
+import java.nio.file.Paths
 
 internal fun List<Pair<TestExecution, String>>.getAndUploadPerformanceMetrics(
-    resultBucket: String
+    args: IArgs
 ) = runBlocking {
     filterAndroidPhysicalDevicesRuns()
         .map { (testExecution, gcsStoragePath) ->
             async(Dispatchers.IO) {
-                testExecution.getPerformanceMetric().upload(resultBucket = resultBucket, resultDir = gcsStoragePath)
+                val performanceMetrics = testExecution.getPerformanceMetric()
+                performanceMetrics.save(gcsStoragePath, args)
+                performanceMetrics.upload(resultBucket = args.resultsBucket, resultDir = gcsStoragePath)
             }
         }
         .awaitAll()
 }
 
+private fun PerfMetricsSummary.save(resultsDir: String, args: IArgs) {
+    val configFilePath = if (args.useLocalResultDir())
```

Comment:
This is only cosmetic but i think smth like 
```
    val configFilePath =
        if (args.useLocalResultDir()) Paths.get(args.localResultDir, "performanceMetrics.json")
        else Paths.get(args.localResultDir, resultsDir, "performanceMetrics.json")
```
is little more readable. WDYT?

Reply:
changed